pkg/slack/filter.go

```diff
@@ -0,0 +1,84 @@
+package slack
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/oursky/github-actions-manager/pkg/github/jobs"
+	"k8s.io/utils/strings/slices"
+)
+
+type messageFilterRule struct {
+	conclusions []string
+	// branches    []string
+	workflows []string
+}
+type MessageFilter struct {
+	whitelists []messageFilterRule
+}
+
+func (rule messageFilterRule) Pass(run *jobs.WorkflowRun) bool {
+	if len(rule.conclusions) > 0 && !slices.Contains(rule.conclusions, run.Conclusion) {
+		return false
+	}
+	if len(rule.workflows) > 0 && !slices.Contains(rule.workflows, run.Name) {
+		return false
+	}
+	return true
+}
+
+func (mf MessageFilter) Any(run *jobs.WorkflowRun) bool {
+	if len(mf.whitelists) == 0 {
+		return true
+	}
+	for _, rule := range mf.whitelists {
+		if rule.Pass(run) {
+			return true
+		}
+	}
+	return false
+}
+
+func (rule *messageFilterRule) setConclusions(conclusions []string) error {
+	conclusionsEnum := []string{"action_required", "cancelled", "failure", "neutral", "success", "skipped", "stale", "timed_out"}
+	// var supportedConclusions, unsupportedConclusions []string
+	var unsupportedConclusions []string
+	for _, c := range conclusions {
+		// if slices.Contains(conclusionsEnum, c) {
+		// 	supportedConclusions = append(supportedConclusions, c)
+		// } else {
+		if !slices.Contains(conclusionsEnum, c) {
+			unsupportedConclusions = append(unsupportedConclusions, c)
+		}
+	}
+
+	if len(unsupportedConclusions) > 0 {
+		if slices.Contains(unsupportedConclusions, " ") {
+			return fmt.Errorf("Do not space-separate conclusions. Use format conclusion1,conclusion2")
+		}
+		return fmt.Errorf("unsupported conclusions: %s", strings.Join(unsupportedConclusions, ", "))
+	}
+
+	rule.conclusions = conclusions
+	return nil
+}
+
+func NewFilter(filterLayers []string) (*MessageFilter, error) {
```

Comment:
Any special reasons for you to return `*MessageFilter`? 

It will be better to return just `MessageFilter` because it is not a stateful object.

Reply:
To be able to return `nil` if there is error.